### DISCUSSION TITLE
Redirects in Icecast

### DISCUSSION THREAD

**Discussion by ivellios1988** [AzuraCast/AzuraCast#6103] [Q&A]
I'd like to be able to redirect my listeners from one Icecast server to another in case all slots are full. My question is, is it possible do to with the version of Icecast used by AzuraCast?

I tried methods described here:
https://icecast.imux.net/viewtopic.php?p=22435&sid=22b24b9065a1df7a1a3d301127cb93ea#22435

When I try to apply <redirect> to all mountpoints it doesn't work, instead, it throws a 403 error message when slots are full and another listener tries to connect.

Any ideas?

**Comment by scysys** [comment]
Icecast's failover only support moving listeners to another mountpoint inside the same icecast server. Moving them from one server to another is not possible with icecast as i know. Im nearly sure 😀

**Comment by ivellios1988** [comment]
After doing some work and asking ChatGPT a ton of questions, I (or maybe I should say 'we' :D) finally came to a working solution. I'll post it here in case someone needs it.

I had to modify my Nginx's configuration file to make Nginx perform redirection when it receives a 403 from Icecast. What I did:

1) add `error_page 403 = @fallback;` at the end of every 'location' section

2) add ```location @fallback {
    rewrite ^(/listen/{URL_STUB}|/radio/{PORT_NUMBER})/(.*)$ http://{path_to_another_server_including_link_to_stream}/$2 redirect;
}```

where:

- {URL_STUB} is a URL Stub I provided on my original station's profile page in AzuraCast
- {PORT_NUMBER} is a port number of my original station
- {path_to_another_server_including_link_to_stream} is a link to another server that, apart from domains, includes a full path to a stream as well
- $2 is my mountpoint

After saving changes in nginx.conf I also changed its permissions to prevent it from being overwritten.

It's necessary to restart Nginx to make changes work.
